server/app/util/filter.py

```diff
@@ -9,18 +14,27 @@ def _convert_variable(datatype, variable):
     :param datatype: type to convert to
     :param variable (string or None): value of variable
     :return: converted variable
-    :raises: ValueError
+    :raises: AssertionError, ValueError
     """
-    try:
-        if variable is None:
-            return variable
-        if datatype == "int":
-            variable = int(variable)
-        elif datatype == "float":
-            variable = float(variable)
+    assert datatype in ["boolean", "categorical", "float32", "int32", "string"]
+    if variable is None:
         return variable
+    try:
+        if datatype == "int32":
+            variable = int32(variable)
+        elif datatype == "float32":
+            variable = float32(variable)
+        elif datatype == "boolean":
+            if variable == "false":
+                variable = False
+            elif variable == "true":
+                variable = True
+            else:
+                raise ValueError
     except ValueError:
```

Comment:
I think you can remove the entire try: except: construct, and just return variable.

Reply:
Fixed.